### PR TITLE
Wpilib units

### PR DESCRIPTION
**Pull request by catr1xLiu**
# Implementing the WPILib units library

use the WPILib units library for APIs and configurations

### CODE REVIEW

File: project/src/main/java/org/ironmaple/simulation/motorsims/requests/PositionVoltage.java [hunk 1]
```diff
@@ -13,9 +13,8 @@ public record PositionVoltage(Angle setPoint) implements ControlRequest {
     public Voltage updateSignal(SimMotorConfigs configs, Angle encoderPosition, AngularVelocity encoderVelocity) {
         Voltage voltage = Volts.of(
                 configs.positionVoltageController.calculate(encoderPosition.in(Radians), setPoint.in(Radians)));
-        Voltage feedforwardVoltage = Volts.of(configs.feedforward.calculate(
-                configs.calculateVelocity(configs.calculateCurrent(encoderVelocity, voltage), voltage)
-                        .in(RadiansPerSecond)));
+        Voltage feedforwardVoltage = configs.feedforward.calculate(
```
Comment (catr1xLiu) [inline project/src/main/java/org/ironmaple/simulation/motorsims/requests/PositionVoltage.java:16]: this is a little unsafe, the feedforward ignores the unit of the input

https://www.chiefdelphi.com/t/wpiliib-simplemotorfeedforward-bug/474469/5

it's better to use the legacy `calculate(double)` method, IMO

File: project/src/main/java/org/ironmaple/simulation/motorsims/requests/VelocityCurrent.java [hunk 2]
```diff
@@ -14,7 +14,7 @@ public Voltage updateSignal(SimMotorConfigs configs, Angle encoderPosition, Angu
         Current current = Amps.of(configs.velocityCurrentController.calculate(
                 encoderVelocity.in(RadiansPerSecond), setPoint.in(RadiansPerSecond)));
         Voltage currentVoltage = configs.calculateVoltage(current, encoderVelocity);
-        Voltage feedforwardVoltage = Volts.of(configs.feedforward.calculate(setPoint.in(RadiansPerSecond)));
+        Voltage feedforwardVoltage = configs.feedforward.calculate(setPoint);
```
Comment (catr1xLiu) [inline project/src/main/java/org/ironmaple/simulation/motorsims/requests/VelocityCurrent.java:17]: same for this one, the legacy calculate(double) method is safer

File: project/src/main/java/org/ironmaple/simulation/motorsims/requests/VelocityVoltage.java [hunk 3]
```diff
@@ -12,7 +12,7 @@ public record VelocityVoltage(AngularVelocity setPoint) implements ControlReques
     public Voltage updateSignal(SimMotorConfigs configs, Angle encoderPosition, AngularVelocity encoderVelocity) {
         Voltage feedbackVoltage = Volts.of(configs.velocityVoltageController.calculate(
                 encoderVelocity.in(RadiansPerSecond), setPoint.in(RadiansPerSecond)));
-        Voltage feedforwardVoltage = Volts.of(configs.feedforward.calculate(setPoint.in(RadiansPerSecond)));
+        Voltage feedforwardVoltage = configs.feedforward.calculate(setPoint);
```
Comment (catr1xLiu) [inline project/src/main/java/org/ironmaple/simulation/motorsims/requests/VelocityVoltage.java:15]: and this

**Review by catr1xLiu — COMMENTED** [review]
most of the code LGTM, only one small problem regarding the usage of feedforward

File: project/src/main/java/org/ironmaple/simulation/motorsims/requests/PositionVoltage.java (hunk 1 above)
Comment (buwunny) [reply project/src/main/java/org/ironmaple/simulation/motorsims/requests/PositionVoltage.java:16]: Sounds good, I had no idea it didn't work as intended.

Comment (catr1xLiu) [reply project/src/main/java/org/ironmaple/simulation/motorsims/requests/PositionVoltage.java:16]: I also didn't know until my feedforward went crazy once 🤣
Anyway, I guess I can do it in the merge

**Pull request merged**